ts-tests/containers/MetaDContainer.ts

```diff
@@ -0,0 +1,168 @@
+import { ethers } from 'ethers';
+import Web3 from 'web3';
+import { JsonRpcResponse } from 'web3-core-helpers';
+import {
+  GenericContainer,
+  Network,
+  StartedNetwork,
+  StartedTestContainer
+} from 'testcontainers';
+import { CHAIN_ID } from '../utils/constant';
+import { HttpProvider, WebsocketProvider } from 'web3-core';
+
+type MetaDNetwork = 'mainnet' | 'testnet';
+
+export interface StartOptions {
+  port?: number;
+  rpcPort?: number;
+  wsPort?: number;
+  timeout?: number;
+  spec?: number;
+}
+
+export class MetaDContainer {
+  static readonly PREFIX = 'metachain-testcontainers-';
+
+  static get image(): string {
+    if (process?.env?.METACHAIN_DOCKER_IMAGE !== undefined) {
+      return process.env.METACHAIN_DOCKER_IMAGE;
+    }
+    return 'ghcr.io/defich/metachain:dc8b57596b57557c3c05cfaa72f98957acb73215';
+  }
+
+  static readonly MetaDPorts = {
+    mainnet: {
+      port: 30333,
+      rpcPort: 9933,
+      wsPort: 9944
+    },
+    testnet: {
+      port: 39333,
+      rpcPort: 19933,
+      wsPort: 19944
+    }
+  };
+
+  genericContainer: GenericContainer;
+  startedContainer?: StartedTestContainer;
+  startOptions?: StartOptions;
+  private network?: StartedNetwork;
+
+  web3!: Web3;
```

Comment:
Is there any particular reason why we need to use `Web3`? From what I see, ethers has a ws provider as well https://docs.ethers.io/v5/api/providers/other/#WebSocketProvider

Reply:
IMO I would rather not use `!` to tell the compiler that this property isn't undefined/not set. 

For eg. in the defibox repo we do something more verbose like

```ts
  get startedTestContainer(): StartedTestContainer {
    if (this.#startedTestContainer === undefined) {
      throw Error('Please start the test container before accessing it.');
    }

    return this.#startedTestContainer;
  }

// we then use the property via the getter
this.startedTestContainer....
```
This way the container _must_ be started before such properties can be accessed.

But I can understand the reasoning here since these containers will __only__ be used for testing (and is also a small part of the overall repo) and will not be shared across other packages/repos.

Reply:
> Is there any particular reason why we need to use Web3? From what I see, ethers has a ws provider as well https://docs.ethers.io/v5/api/providers/other/#WebSocketProvider

yea.. you're right... hmm.. they are quite similar and seems more folks are opt to `ethers`
i think we will rm one in the future and lemme experience their functionality at the moment :sweat_smile:

Reply:
Yep! ethers doesn't have convenience functions like `getHashRate`, but you should be able to make a direct rpc call to the [underlying rpc method](https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_hashrate)